rspace/src/main/scala/coop/rchain/rspace/ReplayRSpace.scala

```diff
@@ -58,325 +54,341 @@ class ReplayRSpace[F[_], C, P, E, A, R, K](store: IStore[C, P, A, K], branch: Br
         val msg = "channels.length must equal patterns.length"
         logF.error(msg) *> syncF.raiseError(new IllegalArgumentException(msg))
       } else
-        syncF.delay {
-          Kamon.withSpan(consumeSpan.start(), finishSpan = true) {
-            val span = Kamon.currentSpan()
-            span.mark("before-consume-lock")
-            consumeLock(channels) {
-              span.mark("consume-lock-acquired")
-              lockedConsume(channels, patterns, continuation, persist, sequenceNumber)
-            }
-          }
-        }
+        for {
+          span <- metricsF.span(consumeSpanLabel)
+          _    <- span.mark("before-consume-lock")
+          result <- consumeLockF(channels) {
+                     lockedConsume(channels, patterns, continuation, persist, sequenceNumber, span)
+                   }
+          _ <- span.mark("post-consume-lock")
```

Comment:
This mark is going to be placed right before the point where the span ought to finish, so it isn't very useful. Why not move it to right after the lock is acquired, so that it shows how long the thread waited for the resource?

Reply:
It's easier to track in zipkin if there is a closing mark.
all the methods have:
`        _          <- span.mark("before-produce-ref-computed")` just after getting the lock.